adafruit_pyportal.py

```diff
@@ -154,7 +154,7 @@ def __init__(self, *, url=None, headers=None, json_path=None, regexp_path=None,
                  image_json_path=None, image_resize=None, image_position=None,
                  caption_text=None, caption_font=None, caption_position=None,
                  caption_color=0x808080, image_url_path=None,
-                 success_callback=None, debug=False):
+                 success_callback=None, esp=None, passed_spi=None, debug=False):
```

Comment:
change `passed_spi` to `external_spi` throughout this PR, it's more descriptive.